lib/idl.ml

```diff
@@ -247,22 +293,36 @@ module GenServerExn () = struct
     | Function (_,f) -> get_error_ty f
     | Returning (_,e) -> BoxedError e
 
+  let rec has_named_args : type a. a fn -> bool =
```

Comment:
It's tricky because it's talking about 'fn', which is local to this module. I talked with @edwintorok yesterday about how to remove a lot of code duplication, which I'd like to do soon.